app/src/auth.ts

```diff
@@ -0,0 +1,192 @@
+import { Auth0Plugin, createAuth0 } from "@auth0/auth0-vue";
+import { App as CapApp } from "@capacitor/app";
+import { Browser } from "@capacitor/browser";
+import { Capacitor } from "@capacitor/core";
+import { type App, watch } from "vue";
+import type { Router } from "vue-router";
+import AuthBrowser from "./util/authBrowser";
+
+const appId = import.meta.env.VITE_CAP_APPID;
+const authDomain = import.meta.env.VITE_AUTH0_DOMAIN;
+
+export type AuthPlugin = Auth0Plugin & {
+    logout: (retrying?: boolean) => Promise<void>;
+};
+
+export default async function setupAuth(app: App<Element>, router: Router) {
+    const app_scheme = `${appId}://${authDomain}/capacitor/${appId}`;
+    const web_origin = window.location.origin;
+    const platform = Capacitor.getPlatform();
+
+    const oauth = createAuth0(
+        {
+            domain: authDomain,
+            clientId: import.meta.env.VITE_AUTH0_CLIENT_ID,
+            useRefreshTokens: true,
+            useRefreshTokensFallback: platform === "web",
+            cacheLocation: "localstorage",
+            authorizationParams: {
+                audience: import.meta.env.VITE_AUTH0_AUDIENCE,
+                scope: "openid profile email offline_access",
+                redirect_uri: platform === "web" ? web_origin : `${app_scheme}/callback`,
+            },
+        },
+        {
+            skipRedirectCallback: true,
+        },
+    );
+
+    // Handle redirects (Save token to local storage)
+    async function redirectCallback(_url: string) {
+        const url = new URL(_url);
+        console.log(url);
+        if (
+            url.searchParams.has("state") &&
+            (url.searchParams.has("code") || url.searchParams.has("error"))
+        ) {
+            if (url.searchParams.has("error")) {
+                const error = url.searchParams.get("error");
+                console.error(error);
+                alert(error);
+            } else {
+                await oauth.handleRedirectCallback(url.toString()).catch(() => null);
+                await Browser.close().catch(() => void 0);
+
+                const to = getRedirectTo() || "/";
+                location.href = to;
```

Comment:
Copilot suggests the following fix:
<img width="805" alt="image" src="https://github.com/user-attachments/assets/982fe589-1115-4f4b-8aac-7637469f3751">

This is not practical, but perhaps we need to do a regex check on the redirect url to make sure it is a normal slug and does not include unwanted 3'rd party addresses (e.g. a different website which will then receive the auth token). I suppose Auth0 will also fail if this is the case?

Reply:
hey @ivanslabbert 

I believe this is an overkill to go into such an extend to try and force security, where auth0 an any case will block the redirect if it is not in the allowed list, 

And if they change the path to the path to on the same domain (app.bcc.africa), they will just end up with at 404.

I don't see it as a security risk, but if you still feel that way then I can see if it is even possible to write a regex that can validate the redirect, since a slug can technically be any thing,